### ISSUE TITLE
Add support for default rke2/k3s version to be configured per rancher version 

### ISSUE THREAD

**Issue by kinarashah** [rancher/rancher#36827]
Currently we can only set one default version in KDM, so all rancher servers reading this data have the same default rke2/k3s version. We need to change this so that starting v2.6.4, they can be different depending on rancher server version. 

For instance, we want k8s v1.21 to stay as default for Rancher v2.6.3 but v1.23 as default for Rancher v2.6.4. 
This might already work because v1.23 is constrained by minRancher version v2.6.4 but we need to confirm the fall back logic works as expected. 

From @snasovich: 
```
The challenge will be that we need to do it in a way that works with 2.6.3 code for which we obviously cannot change.
We do have min/max versions constraints I believe, so 1.23 will be introduced requiring 2.6.4+ I gather, e.g. for 1.22:
https://github.com/rancher/kontainer-driver-metadata/blob/4ecd2cb0ca850a2c7bfa83ed048f3d81179c5faa/channels-rke2.yaml#L370-L372

Not sure what will be the behavior if specified default is not available. I guess it will probably fall back to the latest version available at which time it will be 1.22 which is not desirable for 2.6.3.
``` 

Related Links: 
https://github.com/rancher/rancher/pull/35843 added support for latest channel in Rancher 
https://github.com/rancher/kontainer-driver-metadata/pull/761 added latest channels in KDM

**Comment by snasovich** [comment]
Another scenario we want to check - if we have imported RKE2 cluster in 2.6.3 on k8s 1.21.x, then during edit in the dropdown for k8s version, will 1.22.x be marked as "Experimental" (as 1.21 is no longer default - I believe the "experimental" is added based on the available version being higher than default).

**Comment by snasovich** [comment]
@kinarashah , the reason it's marked regression is due to the fact that if we change the default version in KDM, 2.6.3 is likely to regress in the sense that 1.22 will no longer be marked experimental. This may be stretching the definition of "regression" a bit too far, but I think it's applicable enough to be added.

**Comment by snasovich** [comment]
Regarding https://github.com/rancher/rancher/issues/36827#issuecomment-1065507217, it looks like "experimental" label is controlled by something else on the imported clusters; https://github.com/rancher/dashboard/issues/5409 was initially created based on the understanding it is, but that's not the case. See more information in the linked dashboard issue.

**Comment by jiaqiluo** [comment]
### Root cause
This is a new feature 

### What was fixed, or what changes have occurred

Rancher exposes the following two new endpoints :
- `/v1/management.cattle.io.settings/rke2-default-version`
- `/v1/management.cattle.io.settings/k3s-default-version`

As the names indicate, the value will be the default version for rke2 and k3s.

A new section named `appDefaults` is added to the generated data.json from the KDM server. Please check the following files as examples:
- https://github.com/jiaqiluo/kontainer-driver-metadata/blob/test-app-defaults/data/data.json
- https://github.com/jiaqiluo/kontainer-driver-metadata/blob/test-app-defaults/channels.yaml
- https://github.com/jiaqiluo/kontainer-driver-metadata/blob/test-app-defaults/channels-rke2.yaml

Dashboard UI will use those two endpoints to retrieve the default versions for rke2 and k3s: https://github.com/rancher/dashboard/issues/5545

### Areas or cases that should be tested
- the new endpoints should be available and contain the correct value 
- if the values are changed in the rke-metadata-config, new values should be returned at those two endpoints after refreshing the metadata ( automatically or manually) 
- if the metadata does not contain the `appDefaults` section, the values should be the same as the ones under endpoints `v1-rke2-release/channel` and `v1-k3s-release/channel`, respectively. 
 
### What areas could experience regressions?
All checks around the rke-metadata-config should still pass.

**Comment by jiaqiluo** [comment]
The feature is available in rancher:v2.6-head 50d85b5b1

**Comment by slickwarren** [comment]
tested using v2.6-head (851ff2f):
* the new endpoints should be available and contain the correct value -- pass
  * /v1/management.cattle.io.settings/rke2-default-version
  * /v1/management.cattle.io.settings/k3s-default-version
* if the values are changed in the rke-metadata-config, new values should be returned at those two endpoints after refreshing the metadata ( automatically or manually) -- pass
* if the metadata does not contain the appDefaults section, the values should be the same as the ones under endpoints v1-rke2-release/channel and v1-k3s-release/channel, respectively. -- pass

note -- the dashboard ticket has not yet merged, and will have to be tested at a different time. Therefore the k8s default version(s) when creating/editing an rke2/k3s cluster do not yet reflect the default values at the new endpoints.
